# How taxonomic bias affects abundance measurements {#abundance}

## Rewrite (in progress)

Behind any quantitative measurement lies a mathematical model, either implicit or explicit, that relates the readings of our instruments and the results of our statistical analyses with the biological reality we are measuring.
This section develops a model of microbiome measurement, based on that of @mclaren2019cons, to describe how the actual absolute compositions of microbiome samples are related to estimates of relative and absolute abundance obtained by MGS.
The next section shows how these relationships impact our measurements of differential abundance.

An MGS measurement of a microbiome sample turns a collection of biological source material (cells, viruses, and extracellular DNA) into a collection of taxonomically-assigned reads (or read pairs).
For concreteness and simplicity, we suppose that all assigned reads come from cells and that the ground-truth biophysical quantity we wish to measure is the density of cells (by volume, or by wet or dry mass) belonging to various taxa, or relative abundances directly derived from these densities. 
This particular simplification reflects our own and the literature's prokaryotic bias; however, the features of MGS measurement that we explore also apply to situations (such as studies of viruses, fungi, or environmental DNA) where cell-based measurement may be inappropriate.
We next suppose that there are a set of atomic taxa, which are mutually distinct and within which we assume the properties that determine taxonomic bias are conserved.
For convenience, we identify these atomic taxa with species under a given taxonomic classification system, while acknowledging that whether taxonomic bias is conserved within commonly-accepted species has yet to be investigated in any detail.
Since DA analysis is commonly performed at a variety of taxonomic levels, we consider measurement of species and of synthetic taxa (such as a phylum or a clade in a phylogeny) that consist of multiple species.
We assume that reads assigned to a given taxon and sample really do come from that taxon and sample, but note that taxonomic mislabeling and contamination (from the reagents, lab environment, or other samples) pose further challenges to DA analysis.

- might want to note that we're mostly going to talk as if species-level taxonomic assignment is possible.

Consider the MGS measurement of a particular sample.
For a given taxon, we define the _absolute efficiency_ as the number of reads assigned to that taxon divided by the number of cells in the original sample.
The absolute efficiency depends not only on the ability of the chosen protocol to measure this taxon, but further aspects such as how much the sequencing library was diluted and how deeply the library was then sequenced.
We suppose that the absolute efficiencies of species can be written as a product of a species-specific, sample-independent factor and a species-independent, sample-specific factor.
The species-specific factors reflects the differential ability of a given protocol to measure different species (i.e. taxonomic bias), whereas the sample-specific factor captures the fact that the reads in an MGS measurement are not directly related to the absolute number or density of cells in a sample regardless of which taxa are being measured.
When we consider the _relative efficiencies_—or the ratios among absolute efficiencies—the sample-specific, taxon-independent factor cancels.
Relative efficiencies quantify how much more efficiently different taxa are measured relative to each other.
When we say efficiency unqualified, we mean efficiency relative to some fixed, arbitrary species in the denominator.

For species, the relative efficiencies are sample-independent.
In contrast, the relative efficiency of a synthetic taxon varies depending on the relative densities of its constituent species,
\begin{align}
  \text{efficiency}_T(a) = \frac{\sum_{t \in T} \text{density}_t(a) \cdot \text{efficiency}_t}{\sum_{t \in T} \text{density}_t(a)}
\end{align}
(May need to explain or derive this.)
For example, @mclaren2019cons found that for a particular shotgun-sequencing protocol, the relative efficiency was X between two Firmicute species X and Y;
hence the efficiency of Firmicutes could vary by X depending on which of these two species were dominant in a given sample.

From efficiencies, we can determine the error in relative abundances estimated by our MGS measurement.

- atomic taxa
- synthetic taxa


MGS analysis is commonly done on proportions - the density of a taxon relative to some designated taxonomic whole, such as all prokaryotes.

- give equation def of proportion
- give error in proportion, from efficiency


outline

- Relative-abundance measurements

- Absolute-abundance measurements


questions, thoughts

- should I define absolute efficiency in terms of cells or density?
- need to revise the story of the first paragraphs above.


## Current working version

<!-- TODO: Add an illustration of the bias model in panel 1B -->

```{r fig-abundance, out.width = '65%', fig.cap = '(ref:cap-abundance)', cache = TRUE}
svg_path <- here("figures/absolute-and-relative-abundance.svg")
if (knitr::is_html_output()) {
  knitr::include_graphics(svg_path)
} else if (knitr::is_latex_output()) {
  cowplot::ggdraw() + cowplot::draw_image(svg_path)
}
rm(svg_path)
```

(ref:cap-abundance) Panel A: Metagenomics output is not directly proportional to microbial cell density and so is typically interpreted as _relative abundances_, either as abundances relative to each other (ratio view) or to the total (proportion view). For illustration purposes, the effect of taxonomic bias on the metagenomics output is not shown. Panel B: The effect of taxonomic bias is to distort the relative abundances that are measured by fixed ratios.

_Thoughts for figure 1: ideas for caption: topic sentence can be something about how metagenomics measures biased relative abudances; indicate that A is showing unbiased MGS output; OR: Can I remake this figure such that it shows biased metagenomics output? Consider making label about the model, rather than abundance_

We can understand how taxonomic bias affect relative and absolute abundance measurement using a simplified model of microbiome measurement.
(Details are given in Appendix Model REF)
First, consider the MGS sequencing measurement. 
We start from a sample with a given combination of species in differing absolute abundances, which throughout we take to be in units of density of cells per unit mass.
(For simplicity we ignore issues relating to the sequencing of live versus dead cells and non-cellular DNA.)
Sequencing results in a total number of taxonomically-classified reads that, while not arbitrary, do not straightforwardly or consistently reflect the differences in total microbial density among samples. 
For this reason, these counts are typically only interpreted in relative terms  (Figure 1A)—either relative to each other (i.e., as ratios among taxa) or relative to the total (i.e., as proportions).
Protocols can be implemented to preserve more information about total abundances (See Cruz et al, and Appendix); however, in the main text we focus on the more typical situation where the total count only tells us about the precision of the measured relative abundances and we must make additional measurements or assumptions to recover information about absolute microbial densities.

The effect of taxonomic bias on the relative abundances in a sample is to perturb them by taxon-specific multiplicative factors—the taxa's _measurement efficiencies_—which quantify how efficiently each taxon is converted from cells into taxonomically classified sequencing reads (Figure 1B).
A taxon's efficiency aggregates the bias acrewed during the entire measurement process and can be thought of as a product of efficiencies at various steps in the experimental workflow and other taxon- and protocol-specific factors.
For example, _(Insert Gard and Lacto example, breaking down by extraction, PCR, and copy-number)_.
Only the relative efficiencies affect the measured relative abundances; doubling all efficiencies would not change the mesaured ratios or proportions.
Therefore we define the efficiencies as relative to a chosen reference taxon (or set of taxa); here we'll take to be that of the least-efficiently measured taxon being considered.

- TODO: Somewhere Give 1-sentence summary of the model - absolute information is lost, and measured relative abundances are perturbed by taxon- and protocol-specific factors. (In the figure caption and/or start of this section and or right here as a summary/resolution)
- Think about How what I write below depends on whether I'm taking this equation to in effect be a statement of the model, or the model is what I illustrated.
- Should define what I mean by Taxon, and address that we're making a big assumption that efficiency of a taxon is fixed (including doesn't vary from lower-level variation)
- A footnote or "Details" box can discuss what happens when there is genotypic or non-genotypic variation in efficiency within a taxon.


## Relative-abundance measurements

By this model, the ratios in the read counts of taxa, compared to the ratios among their cell densities, are multiplied by taxon- and protocol-specific factors (Figure, MWC).
\begin{align}
  (\#eq:ratio-err)
  \frac{\text{reads}_{i}(s)}{\text{reads}_{j}(s)}
  &= \frac{\text{density}_{i}(s)}{\text{density}_{j}(s)} \cdot \frac{\text{efficiency}_{i}}{\text{efficiency}_{j}}.
\end{align}
But what about the estimated proportions, estimated by dividing the taxon read counts by the total?
The estimated proportion of a taxon $i$ in a sample $s$ under this model equals (@mclaren2019cons, Box)
\begin{align}
  (\#eq:prop-err)
  \hat{\text{prop}}_{i}(s) 
  &= \text{prop}_{i}(s) \cdot \frac{\text{efficiency}_{i}}{\text{mean efficiency}(s)}
\end{align}
where $\text{mean efficiency}(s)\equiv \sum_{j}\text{prop}_j(s)\cdot \text{efficiency}_j$ is the average efficiency of cells in the sample.
The mean efficiency in the denominator of \@ref(eq:prop-err) was omitted in models of bias prior to that of @mclaren2019cons, but naturally arises from the requirement that the estimated proportions sum to 1 and captures the competition-to-be-measured that this sum-to-1 constraint implies ([Details](#details:relative)).
Intuitively, Equation \@ref(eq:error) says that the estimated proportion of taxon $i$ will be too large when its measurement efficiency is greater than the average efficiency of cells in the sample.

<details id="details:relative">
<summary style="font-weight:bold;font-size:120%">Details: Taxonomic bias and relative-abundance estimates</summary>
::: {.rmdnote}
<!-- ### Details: Taxonomic bias and relative-abundance estimates {-} -->

**Proportions:** The measured proportion of a taxon $i$ in a sample $s$, defined as the ratio of the taxon's density to the total density of all taxa in the sample, is
\begin{align}
  (\#eq:prop-def)
  \text{prop}_{i}(s) \equiv \frac{\text{density}_{i}(s)}{\text{total density}(s)}.
\end{align}
The standard estimate for the proportion of taxon $i$ is the proportion of its reads,
\begin{align}
  (\#eq:prop-est)
  \hat{\text{prop}}_{i}(s) \equiv \frac{\text{reads}_{i}(s)}{\text{total reads}(s)}.
\end{align}
These definitions imply that both the actual and the estimated proportions each sum to 1.
Our model stipulates that within a given sample $s$, the estimated and actual proportions are related by 
\begin{align}
  \hat{\text{prop}}_{i}(s) = c(s) \cdot \text{prop}_{i}(s) \cdot \text{efficiency}_{i}
\end{align}
for some taxon-independent coefficient $c(s)$.
The requirement that the estimated proportions sum to 1 implies that 
\begin{align}
  c(s) = \sum_j \text{prop}_{j}(s) \cdot \text{efficiency}_j = \text{mean efficiency}(s).
\end{align}

**Ratios:** The actual ratio of taxon $i$ to taxon $j$,
\begin{align}
  (\#eq:ratio-def)
  \text{density}_{i/j}(s) \equiv \frac{\text{density}_{i}(s)}{\text{density}_{j}(s)}
\end{align}
is estimated by the ratio of reads assigned to each taxon,
\begin{align}
  (\#eq:ratio-est)
  \hat{\text{density}}_{i/j}(s) \equiv \frac{\text{reads}_{i}(s)}{\text{reads}_{j}(s)} = \frac{\hat{\text{prop}}_{i}(s)}{\hat{\text{prop}}_{j}(s)}.
\end{align}
The error under multiplicative bias is
\begin{align}
  (\#eq:ratio-err)
  \hat{\text{density}}_{i/j}(s) = \frac{\text{density}_{i}(s)}{\text{density}_{j}(s)} \cdot
  \frac{\text{efficiency}_{i}(s)}{\text{efficiency}_{j}(s)}.
\end{align}

:::
</details>

<!-- begin figure -->

```{r fig-error-proportions, out.width = '90%', fig.cap = '(ref:cap-error-proportions)', cache = TRUE}
svg_path <- here("figures/error-proportions.svg")
if (knitr::is_html_output()) {
  knitr::include_graphics(svg_path)
} else if (knitr::is_latex_output()) {
  cowplot::ggdraw() + cowplot::draw_image(svg_path)
}
rm(svg_path)
```

(ref:cap-error-proportions) Top row: Error in proportions measured by MGS in two communities that contain different relative abundances of three taxa. Bottom row: Error in the estimated fold-change in the third taxon that is derived from these measurements.

<!-- end figure -->

This competitive aspect of proportional measurement creates a notable feature of the errors: They depend on the composition of the sample.
The same taxon, subject to the same measurement efficiency, will have a different error in its estimated proportion in different samples depending on the taxonomic composition of the sample (Figure 2A+B).
A taxon will be overestimated in samples where its efficiency is greater than that of the sample average, and underestimated in samples where its efficiency is less than the sample average (e.g. Taxon 3 in 2A vs 2B).
This difference in the consistancy of ratios versus proportions propagates to ratio- and proportion-based absolute abundance estimates and, as we describe below, is a key tool for understanding how taxonomic bias affects different DA analyses.

<!-- discussion of competition; tentative positioning -->

_The below discussion makes more sense if the goal as density/AA inference, and not proportion inference. Perhaps should split out some if it into the AA section._

Why does the sequencing outcome associated with one taxon depend on the rest of the sample?
We can think of proportional measurement as a competition among all the cells in the sample for a limited pool of 100 percentage points, for which
the measurement efficiencies determine each cell's ability to compete.
It is therefore a cell's efficiency relative to its average competitor (all the cells in the sample) that determines how many percentage points it receives.
This account highlights that it is the competition is the problem---if we were able to measure taxa/cells in a taxonomically biased but non-competitive manner, then we could get consistent errors. We cannot remove competition simply by using reads counts instead of proportions, since competition is intrinsic to sequencing experiments, due to limits in the yield during DNA extraction, PCR, and sequencing that occur in sequecing experiments.

_Perhaps say something about ratios - since cells or taxa in the sample are subject to the same average competitor, the effect of the average competitor cancels out._

Our ability to design-out these limits is limited and rarely attempted (ref Cruz for counter example).
But we might overcome competition with other types of measurements, particularly those that target individual taxa. For example, ddPCR can be applied directly to an unextracted sample to count individual cells of a given taxon, before any competition associated with the sequencing protocol has occurred. Targeted qPCR quantifies a marker (typically the 16S gene) in the extracted DNA, and so is only affected by competition during the DNA extraction step. These measurements are still subjected to taxonomic bias, but the lack or reduction in competition means that the error due to taxonomic bias is likely to be relatively consistent. For this reason, these measurements can serve as useful control measurements for correctly for the variation in the sample mean efficiency in the sequencing measurement.

## Absolute-abundance measurements

- P: Motivation for AA
However, these relative-abundance measures leave ambiguous what is happening to the absolute cell densities of taxa.
- e.g. of proportion decrease coming from different explanations
- e.g. of ratio varying having different explanations.

AA methods have generally been motivated, developed, and applied ignoring taxonomic bias and assuming the estimated proportions are correct.

The estimation procedures used by AA methods can be split into those that use the proportions estimated from the sequencing reads, and those that use ratios.

Proportion-based estimation involves the intuitive procedure of multiplying the estimated proportions by an independent estimate of the total cell density in the sample.
(The total-density estimates can be obtained by a wide variety of means, reviewed in APPENDIX.)
By this approach, the estimated density of a taxon $i$ in sample $s$ is simply
\begin{align}
  (\#eq:density-prop-est)
  \hat{\text{density}}_{i}(s) 
  = \hat{\text{prop}}_{i}(s) \cdot \hat{\text{total density}}(s).
\end{align}
This estimate equals the true density if the estimated proportions and total density are both accurate.
Accounting for the error in the estimated proportions due to taxonomic bias (by substituting in EQ-REF) and potential error in the estimated total density gives
\begin{align}
  (\#eq:density-prop-err)
  \hat{\text{density}}_{i}(s) 
  &= \text{density}_{i}(s) \cdot \frac{\text{efficiency}_{i}}{\text{mean efficiency}(s)} \cdot \text{error in }\hat{\text{total density}}(s).
\end{align}
Densities estimated from proportions are subject to the same bias-induced error as those proportions, which depends on sample composition through the sample's mean efficiency.

Error in the total density estimate is also typically ignored but may be considerable (Ref future box) and can worsen or improve the error in DA analysis; we return to its implications in the context of considering how these errors impact DA analysis.

Ratio-based density estimation embodies the fact that, without knowing the total abundance, we can regain absolute abundance information if we can determine the abundance of one or more _reference taxa_.
Given an estimate of the density of reference taxon $r$, we can estimate the density of taxon $i$ by multipling the ratio of $i$ to $r$ in the reads by the density of $r$,
\begin{align}
  \hat{\text{density}}_{i}(s) 
  &= \frac{\text{reads}_{i}(s)}{\text{reads}_{r}(s)} \cdot \hat{\text{density}}_{r}(s);
\end{align}
So far this approach has been mainly used with spike-in experiments, in which an extraneous taxon is added in a known (and typically constant) abundance to each sample so that it can serve as the reference taxon.
It has also been used by methods that computationally determine a set of "housekeeping taxa" that they then assume have a constant abundance across samples.
Another potential application is to use naturally occurring taxa that we have biological reason to believe are constant, as has been argued for host and organelle-derived reads in plant and animal microbiome studies.
Moreover, EQREF makes clear there is no need for the reference taxa to be constant or known in advance, so long as we can measure its abundance: Hence the reference taxon can also be a naturally occurring taxon of varying abundance whose density we have estimated using a method such as ddPCR directly on cells or (q/dd)PCR on the extracted DNA.

The error in this estimate, accounting for the error in the estimated ratio due to taxonomic bias (EQ-REF) and potential error in the estimated density of the reference taxon, is described by
\begin{align}
  \hat{\text{density}}_{i}(s) 
  &= \text{density}_{i}(s) \cdot \frac{\text{efficiency}_{i}}{\text{efficiency}_{r}} \cdot \text{error in } \hat{\text{density}}_{r}(s)
\end{align}
Again, the properties of the corresponding relative-abundance estimate carries over, so that the error induced by taxonomic bias in the metagenomics measurement is independent of sample composition.
The reference density estimates are subject to error, but we can reasonably expect this to be more consistent across samples (see box), leaving the overall error relatively constant.

A warning note: The distinction between ratio and proportion-based estimation depends on whether an equation resembling X or Y is used for the final density estimates, and does not cleanly align with which empirical method is used to obtain absolute-abundance information. In particular, some studies that use spike-ins and most that use host or organelle reads to obtain absolute abundances do not follow this approach: Instead, these studies use the ratio of total reads of focal taxa to the reference reads as a proxy for total density to use in forming proportion-based estimates (EQREF). (For studies using host reads, the total density estimate may come from WGS sequencing and the estimated proportions from 16S sequencing.) 
Alhough there may be other advantages to this approach, the above observations suggest that it may be more susceptible to taxonomic bias in the sequencing measurement than using a ratio-based approach on the same underlying data.

**Update:** Actually, the two methods are equivalent when the total density and proportion are estimated using the same reads.
Let $B$ be all bacteria and $H$ be the host.
Consider using WGS reads to estimate the absolute abundance of bacterial taxon $i$,
\begin{align}
  \hat{\text{density}}_{i}(s) 
  &= 
  \underbrace{\frac{\text{reads}_{B}}{\text{reads}_{H}}}_\text{est'd bact'l density} \cdot 
  \underbrace{\frac{\text{reads}_{i}}{\text{reads}_{B}}}_\text{est'd prop'n}
  = \frac{\text{reads}_{i}}{\text{reads}_{H}}
\end{align}
So the problem/difference arises when we the proportion is estimated either by a different technology (e.g. when density is estimated via WGS and proportion is estimated by 16S sequencing) or the two estimates going into the above are not just ratios of reads.
For example, ratios of bacterial to host reads could be used for the density estimate, while the proportion is estimated by Metaphlan.
