# How taxonomic bias affects differential-abundance analysis

_The current text generally assumes that taxa are atomic._

The current practice of ignoring bias in DA and other analyses of the differences betwen samples is sometimes justified by the idea that the error in measurements of in individual samples should not affect the measured differences between samples, so long as they have been subjected to the same taxonomic bias.
This argument is predicated on an assumption that consistent taxonomic bias creates consistent errors in the quanity of interest.
The previous section showed that while this may be true for ratio-based abundance measurements, it is not true for proportion-based measurements.
This section formally describes how the errors in the measurements of individual samples translate into errors (or lack thereof) in differential abundance analysis of (log) fold changes in proportions, ratios, and densities.

## Change between a pair of samples

Before considering the sterotypical many-sample DA analysis, it is instructive to 
consider simplest analysis of differential abundance: the estimation of fold changes in abundance between a pair of samples.
This case is relevant for understanding common visualizations for comparing abundances across individual samples, such as the ubiquitous proportion bar plot and abundances-through-time trajectories within a single host or environment.
Just as important, it helps to bridge intuitions from the single-sample results of the previous section to the many-sample case.

<!-- P: error in FC proportions -->
The composition-dependent effect of bias on fold error in proportions leads to error in fold-change estimates that is proportional to the inverse change in mean efficiency (REF Figure 2B).
From the error in individual sample \@ref(eq:prop-err), it follows that the estimated fold change in proportion of taxon $i$ from sample $s$ to sample $t$ is
\begin{align}
  (\#eq:prop-fc-err)
\underbrace{\frac{\hat{\text{prop}}_{i}(t)}{\hat{\text{prop}}_{i}(s)}} _\text{estimate}
  &= \frac
    {\text{prop}_{i}(t) \cdot \cancel{\text{efficiency}_{i}} / {\text{mean efficiency}(t)}}
    {\text{prop}_{i}(s) \cdot \cancel{\text{efficiency}_{i}} / {\text{mean efficiency}(s)}}
\\[0.5ex]
  &= 
  \underbrace{\frac{\text{prop}_{i}(t)}{\text{prop}_{i}(s)}}_\text{actual}
  \cdot 
  \underbrace{\left[\frac{\text{mean efficiency}(t)}{\text{mean efficiency}(s)}\right]^{-1}}_\text{error}
  .
\end{align}
The sample-independent efficiency factor of the error cancels, but the sample-dependent mean efficiency does not, leaving an error equal to the inverse of the change in the mean efficiency from $s$ to $t$.
In the example in Figure 2, the mean efficiency decreases by 2.64X from 8.33 to 3.15, creating an artificial fold increase of 2.64X in the estimated fold changes.
This same error applies to all taxa, but its consequences differ depending on the taxon's actual fold change.
For Taxon 1, which increases by 2.3X, we estimate larger increase of 6.5X; For Taxon 2, which decreases with a FC of 0.15X, we see a lesser decrease of 0.44X; For taxon 3, which decreases with a FC by 0.6X, we see an _increase_ of 1.75X.
Thus for the same pair of samples we can see errors of increased magnitude, errors of decreased magnitude, and errors of changed direction (or sign) for different taxa depending on how the magnitide and direction of the true change compares to that of the mean efficiency. Finally, any taxon whose proportion is constant will appear to vary precisely as the inverse mean efficiency.

TODO: Fig 2 Add numbers for the other taxa.

<!-- P: error in proportion-based AAs (revise) -->
Taxonomic bias causes the same error in the estimated fold changes in proportion-based density estimates.
A similar calculation from the error in individual density estimates \@ref(eq:density-prop-err) shows that the error in the fold change of a taxon $i$ is given by the analogous formula to \@ref(eq:prop-fc-err), with density subsituted for proprotion.
The same types of errors (in magnitude and sign) can arise (Figure 3), now depending on how the magnitide and direction of the true change in density compares to that of the mean efficiency (Figure 3 TBD).
Maybe text:
- Alternative view of what is doing on, using the Lacto/Gard example, where a rarer but lower efficiency taxon grows. By first view, the mean efficiency decreases, causing an upward bias in the FC estimates. By the second view, the gains in total density are incorrectly attributed to the more efficiently measured taxon.
- This figure and its discussion should relate these two views. Perhaps the first mention of AA estimates should as well.
- Note: Whereas sign errors in proportion FCs require the involvement of at least three taxa (TODO: explain in footnote), sign errors in proportion-based density FCs can occur even with just two taxa.

NOTE: Having a figure and example specifically for density is not necessary for this draft.


<!-- P: error in FC ratios and ratio-based AAs -->
(In contrast to proportion-based estimates) Because consistent taxonomic bias creates a consistent error in ratio-based estimates, the error completely cancels, leaving accurate fold changes.
From \@ref(eq:ratio-err), the estimated fold change in the ratio of taxon $i$ to taxon $j$ from sample $s$ to sample $t$ is
\begin{align}
  (\#eq:ratio-fc-err)
  \underbrace{ \frac{{\text{reads}_{i}(t)}/{\text{reads}_{j}(t)}} {{\text{reads}_{i}(s)}/{\text{reads}_{j}(s)}} }_\text{estimate}
  &= 
  \frac{{\text{density}_{i}(t)}/{\text{density}_{j}(t)}} {{\text{density}_{i}(s)}/{\text{density}_{j}(s)}}
  \cdot
  \frac{\cancel{\text{efficiency}_{i}}/\cancel{\text{efficiency}_{j}}} {\cancel{\text{efficiency}_{i}}/\cancel{\text{efficiency}_{j}}}
  \\[0.5ex]&= 
  \underbrace{ \frac{{\text{density}_{i}(t)}/{\text{density}_{j}(t)}} {{\text{density}_{i}(s)}/{\text{density}_{j}(s)}} }_\text{actual}
  .
\end{align}
(Describe FCs in Figure 2 and Figure 3 density e.g., seeing the correct FCs.)
(_Need to make plain in the equation that this is the estimated fold change in the density of taxon i; perhaps have notation for this._)

### Synthetic or higher-level taxa

_Some of these concepts would best be introduced in the previous secton on abundances.
Alternatively, we might want to separate them into a box or appendix on multiplicative aggregation._

<!-- P: Synthetic taxa intro -->

Frequently, relative- and absolute-abundance estimation are used to infer fold changes in higher-level taxa.

- Visual or statistical estimation of changes in phylum abundance is a ubiquitous form of analysis in bacterial microbiome studies.
- @chng2020meta - use fold changes in the ratio of bacterial to host reads to estimate the change in bacterial biomass during mouse antibiotics experiment
- @rao2021mult - use a spike-in to infer changes in total domain abundance for each of archaea, bacteria, and fungi.
- @karasov2020ther - ratio of microbial to host reads to infer changes in microbial biomass 

We know that efficiencies vary widely within domains and also has been observed to vary within phyla and even within species.
What are the implications of this within-taxon variation for infered fold changes?

<!-- P: Synthetic taxa results -->

Unfortunately, the bias invariance of ratio-based inference of species need not apply to ratios involving higher-level taxa.
The estimated fold change in the proportion, ratio, or density of a synthetic taxon $T$ has additional multiplicative error equal to its fold change in efficiency.
This error can be substantial when there is a substantial turnover from dominance by high- to low-efficiency species (or vice versa).
(todo: ref relevant equation from previous section)

An alternative approach to creating synthetic quantitative features was develoepd in the Compositional Data Analysis field that retains the so-called perturbation invariance property of atomic features.
These "balances" are constructed by multiplying, rather than adding, the abundances of the atomic elements (equivalent to adding the abundances after log transformation instead of before) (@vandenboogaart2013anal).
The changes in balances retain bias invariance (@mclaren2019cons Appendix 1, and REF Appendix).
Balances have been used to construct candidate disease biomarkers (@riverapinto2018bala, @quinn2020inte), develop phylogenetic community distance measures (@silverman2017aphy), and look for clades associated with traits (@washburne2017phyl). 
But they face from conceptual and practice issues relating to the sparsity of microbiome data. 
Perhaps more important, products of species abundances do not obviously map onto the ecological notion of abundance and hence do not seem to fully substitute for attempting to infer the changes in taxon abundance as conceived in the usual sense of additive aggregation.
A benefit of products, however, is that they are also robust to the fact that we may not know the appropriate ecological conversion rate for different species; for example, cells of one clade might be twice as impactful per cell on a host or  ecosystem as cells from another clade.
Grappling with these issues remains an important goal for future theoretical and empirical research.

## Regression over many samples

Formal DA analysis is typically used to measure average patterns across many samples, such as how the abundance of a taxon varies with some covariate of interest.
The covariate may be discrete, such as whether the sample is from a healthy or sick person, or it may be continuous, such as a measure of space, time, or temperature.
Often these analyses can be framed as a regression problem.
For example, we might hypothesize that the log density of an individual taxon $i$ changes with a variable $x$ according to the simple linear regression, 
\begin{align}
  E\left[\log \text{density}_{i} \mid x \right] = a_{0} + a_{1}x
\end{align}
where $x$ is either continuous (e.g., sediment depth) or binary (e.g., $x=1$ for treated patients and $x=0$ for controls).
For a continuous covariate, these coefficients are the intercept, $a_0$, and the slope, $a_1$, of a line.
For a binary covariate, they are the average abundance in the first condition and the change in average abundance of the second condition.
How does taxonomic bias in the measurements of individual samples affect our estimate of the regression, and in particular the slope or difference term $a_1$, for log proportions, ratios, and densities?

_NOTE: Consider making the above abundance instead of density, and say that it could be proportion, ratio, or density_

### Proportion-based

P: Results for proportions and for proportion-based density, and 
- Equations for both; ref details box and SI for details and derivations
- The sample-dependent term is absorbed by the model intercept, which is typically not of interest to the researcher. - connect to canceling of efficiencies in FC result
- Error in slope equal to inverse slope in mean efficiency - connect to FC result
- Intuitive summary in terms of the correlation / covariation with the covariate 
    - When covaries - statistical bias
    - When doesn't covary - systematic errors between pairs of samples acts as random error in the regression -> reduced precision and power to detect differences.
- as w/ FCs, whether the error is significant depends on magnitude of true slope relative to slope of ME.
- in addition, depends on the SEs of the estimated slope

P: Discuss figure showing results of real and/or simulated example
- shared LFC error -> sign or (2 types of) magnitude errors for different taxa
        - if use leopold e.g., edit to show diff taxa on sample samples
- increase in residual variation and hence standard errors (NEED TO SHOW)
- old text
    - Notably, the error in an absolute sense is the same for all taxa—the slope is always reduced by the same amount—but the implications differ for various taxa depending on their slopes, causing magnitude or sign errors depending how the coefficient derived from the (unknown) true abundances compares to that of the mean efficiency (Figure \@ref(fig:regression-example)).
    
#### Figure

```{r regression-example, fig.cap = '(ref:regression-example)'}
svg_path <- here::here("figures/leopold2020host-example.svg")
if (knitr::is_html_output()) {
  knitr::include_graphics(svg_path)
} else if (knitr::is_latex_output()) {
  cowplot::ggdraw() + cowplot::draw_image(svg_path)
}
rm(svg_path)
```

(ref:regression-example) **Taxonomic bias distorts multi-sample differential abundance inference when the mean efficiency of samples is associated with the covariate of interest.**

TODO: Add SI figure showing the distribution of mean efficiencies and the relationship with Mel that goes with this figure

Ideal figure perhaps shows how observed = actual + eff - mean eff, and individual taxa cases of sign and magnitude errors, and the distributional shift of all taxa, with the specific examples pulled from this distribution


```{r lfc-distribution, fig.cap = '(ref:lfc-distribution)', out.width = "60%"}
svg_path <- here::here("figures/illustrations/lfc-distribution.svg")
if (knitr::is_html_output()) {
  knitr::include_graphics(svg_path)
} else if (knitr::is_latex_output()) {
  cowplot::ggdraw() + cowplot::draw_image(svg_path)
}
rm(svg_path)
```

(ref:lfc-distribution) **Taxonomic bias causes magnitude and sign errors in differential-abundance estimates.** In the absence of other types of errors, consistent taxonomic bias causes shift in the estimated log fold changes (LFC) or slopes in a differential-abundance analysis that is shared by all taxa, creating magnitude or sign errors depending on the true LFC. 

### Ratio-based

1 short P explaining how it is no surprise that in ratio based, the error is absorbed by the intercept and the slopes remain accurate.
- (summary of previous) leads to Taxonomic bias does not affect ratio-based DA analysis
- include consideration of error in density estimate

## Caveats and other considerations

- Other error sources and violations of this idealized model
    - inconsistent error in density estimate, could be correlated or not with the covariate
    - for DNA based: non-linearity of extraction
    - contamination
- implications of these in our above results
    - proportion-based: won't get perfect error sharing across taxa
    - ratio-based: won't get perfect canceling
    - -> can limit the effectiveness of our 'solutions' below.

For this version, can just say this, and in future versions I can be more concrete as I think through more .

### Error in density estimates

In reality, there will be some error in the total density.
If this error is correlated with the mean efficiency of the sequencing measurement, it can actually reduce the error in the fold change estimate (See Box).

Possible continuations of this P:
1) Such a correlation might arise through DNA extraction bias when the total density is measurements of total abundance from DNA, such as florescent-based DNA quantification or qPCR of a universal 16S marker.
In this case, a perfectly linear but taxonomically biased extraction protocol will have a DNA yield that is proportional to the true density and the average absolute extraction efficiency of its cells.
16S concentration in the extracted DNA will be proportional to the average produce of extraction efficiency with 16S copy number.

2) work from old, and just use qPCR as a focal example.

- Could also mention live vs dead cells and cell-free DNA as another bias source, giving a ref to below article
- also mention primer bias; again can be helpful if shared. Obvious case is if both qPCR and seq primers are not getting certain taxonomic groups, that might be picked up by an alt approach like bulk DNA or flow cytometry.

Next P:
Shows subtleties and utility of thinking through an explicit model of the measurement process. 
When we do not attempt to correct for bias, DNA-based estimate of total might be better for proportion-based density estimates despite giving a poor reflection of the total density.
For example, (Lacto/Gard example) -> extraction and copy number bias gibe spurious results in total, but good results in individual taxa.
(Have a details box that does this as a worked example)
However, the non-linearity of DNA extraction may be a major worry in some cases and does not affect direct cell-density measurements.
Therefore, alternative options from the solutions section below may be preferable.

Old text to use: 

Interestingly, taxonomic bias may actually make qPCR a better method for absolute quantification than cell counting for 16S rRNA gene sequencing experiments.
16S rRNA gene qPCR measures the concentration of 16S rRNA gene copies in the extracted DNA and is therefore necessarily affected by three large sources of bias in 16S rRNA gene experiments: extraction, amplification, and copy-number variation.
Yet these biases are shared by the sequencing measurement.
Though they make qPCR measurements a bad proxy for total cell density, when used for absolute differential abundance inference the shared bias in the qPCR and sequencing measurements can offset each other, leading to estimated fold changes that are relatively unaffected by taxonomic bias.
(Jian, Salonen, and Korpela 2021 suggests some of these ideas and APPENDIX gives a mathematical justification.)
If common primers are used by qPCR and 16S rRNA gene sequencing, the bias due to primer mismatches and other sources of variation in amplification could even be accounted for in this manner.
Yet ideally we would still be able to estimate and correct remaining error due to unshared bias or to mechanisms, such as saturation during DNA extraction, that could break the assumed proportionality between qPCR measurements and total abundance.
In this case, ...
