#+title: Outline of differential-abundance-theory manuscript
#+options: toc:nil
#+latex_class_options: [12pt]
#+latex_header: \usepackage[total={6in, 8.5in}]{geometry}
#+latex_header: \usepackage{cancel}
#+latex_header: \hypersetup{colorlinks=true, linkcolor=blue}
* Introduction
Paragraph outline:
- P: Differential-abundance (DA) analysis is the most basic and ubiquitous form of microbiome analysis applied to MGS data.
- P: Taxonomic bias makes the MGS measurements that DA analysis are based on inaccurate
- P: It is commonly thought that consistent bias should not affect cross sample comparisons, but McLaren et al 2019 showed that consistent bias can lead to spurious results in DA analysis.
- P: Second reason bias is ignored is lack of tractable solutions
- P: Objective and summary of the present article: Explain effect of consistent bias on standard relative and absolute DA analysis, describe scenarios where bias is and isn't likely to cause spurious results, and present several solutions.

To adjust the current [[https://da.mikemc.cc/revision/introduction.html][intro in the 'revision' branch]] to fit this outline,
1. move solutions paragraph up one, and
2. write an objective/summary paragraph as last, which combines points from the current last 2 paragraphs
* How taxonomic bias affects abundance measurements
** Model
- State model of sequencing-based measurement of relative and absolute abundance
  + Define model at the species level; stipulate that species have consistent properties across samples w.r.t. taxonomic bias.

Model definition: The reads and density of a taxon are related by a species-specific factor, its /relative efficiency/, and a sample-specific factor, the /sequencing effort/ for that sample.
\begin{align}
  \text{reads}_{i}(a)
  = \text{density}_{i}(a) \quad \cdot
    \underbrace{\text{efficiency}_{i}}_{\substack{\text{species specific,} \\  \text{sample independent}}}
    \cdot \quad
    \underbrace{\text{sequencing effort}(a)}_{\substack{\text{species independent,} \\  \text{sample specific}}}
\end{align}
Discuss with Ben: Is 'sequencing effort' a good term for the sample-specific factor?
** Error in relative abundance (ratios and proportions)
- Error in ratio of two species
#+name: eq:ratio-err
\begin{align}
%  (\#eq:ratio-err)
  \frac{\text{reads}_{i}(a)}{\text{reads}_{j}(a)}
  &= \frac{\text{density}_{i}(a)}{\text{density}_{j}(a)} \cdot \frac{\text{efficiency}_{i}}{\text{efficiency}_{j}}.
\end{align}

- Error in proportion
  - Define proportion as density of species $i$ divided by total density for all species in the taxonomic domain under study (e.g. all fungi or all prokaryotes).
- Intermediate step: The total reads equal
\begin{align}
  \text{total reads}(a)
    = \text{total density}(a) \cdot \text{mean efficiency}(a) \cdot \text{sequencing effort}(a),
\end{align}
where $\text{mean efficiency}(a)\equiv \sum_{j}\text{prop}_j(a)\cdot \text{efficiency}_j$ is the average efficiency of cells in the sample.
- So the error in the proportion of species $i$ estimated by $\text{reads}_i(a) / \text{total reads(a)}$ is
\begin{align}
%  (\#eq:prop-err)
  \hat{\text{prop}}_{i}(a)
  &= \text{prop}_{i}(a) \cdot \frac{\text{efficiency}_{i}}{\text{mean efficiency}(a)}.
\end{align}
- This equation is analogous to Equation [[eq:ratio-err]] (ratio case), with the denominator taxon is the entire taxonomic domain and the relevant efficiency is the weighted average of all species.
- Intuitively, a species is measured as having a larger-than-actual proportion when it is more efficiently measured than average cell in the sample.
   - Same species can be over or under measured in terms of its proportion.
- Discuss top row of Figure [[fig:error-proportions]]

#+name: fig:error-proportions
#+attr_org: :width 500px
#+caption: Top row: Error in proportions measured by MGS in two communities that contain different relative abundances of three taxa. Bottom row: Error in the estimated fold-change in the third taxon that is derived from these measurements.
[[file:figures/error-proportions.png]]
# [[file:figures/error-proportions.svg]]

- Key takeway: The fold error in ratios is consistent, but the fold error in proportions varies with the mean efficiency.
** Error in absolute abundance (cell density)
*** Introduce absolute abundance estimation
  + Various methods exist, but all can be split into whether they use the species proportions or ratios from MGS to estimate species densities.
*** Proportion-based density estimate
Method: Estimate total density; multiply estimate of total by MGS proportions.
Error in proportion-based density estimate is
\begin{align}
  % (\#eq:density-prop-err)
  \hat{\text{density}}_{i}(a)
  &= \hat{\text{prop}}_{i}(a) \cdot \hat{\text{total density}}(a)
\\&= \text{density}_{i}(a) \cdot \frac{\text{efficiency}_{i}}{\text{mean efficiency}(a)} \cdot \text{error in }\hat{\text{total density}}(a).
\end{align}

Discuss with Ben: Whether to include and discuss the error in the total density estimate.
*** Ratio-based density estimate
Given an estimate of the density of reference taxon $r$, we can estimate the density of taxon $i$ by multipling the ratio of $i$ to $r$ in the reads by the density of $r$,
Error in ratio-based density estimate is
\begin{align}
  \hat{\text{density}}_{i}(a)
  &= \frac{\text{reads}_{i}(a)}{\text{reads}_{r}(a)} \cdot \hat{\text{density}}_{r}(a)
\\&= \text{density}_{i}(a) \cdot \frac{\text{efficiency}_{i}}{\text{efficiency}_{r}} \cdot \text{error in } \hat{\text{density}}_{r}(a)
\end{align}

To discuss with Ben: Whether to include and discuss the error in the reference taxon
*** Conclusion
Fold error is consistent in ratio-based, but not proportion-based abundance estimates, for which the fold error can vary w/ the mean efficiency.
* How taxonomic bias affects differential-abundance analysis
- Focus on log fold changes in proportions, ratios, and cell density because 1) these are very common metrics of DA that 2) have the most direct ecological interpretations (via the processes of exponential growth and death) than any DA measure and 3) are the best-case/most-intuitive scenario for the effect of bias canceling out.
** Change between a pair of samples
Outline:
- P: Intro
- P: Error in FC's in proportions between two samples
  + Equation (below)
  + Explain why error only partly cancels
  + Three types of errors: magnitude increase, magnitude decrease, and sign, depending on true change vs. change in mean efficiency
  + Illustrate with Figure [[fig:error-proportions]] 2nd row
- P: (perhaps 1 sentence) Same issue for proportion-based density: error in FCs equal to inverse change in FC in mean efficiency.
- P: FCs in ratios and ratio-based density estimates.
  + (can explain w/o any new equations) Effect of bias completely cancels, leading to correct FC

Equation for estimated fold change in proportion of species i from sample a to sample b:
\begin{align}
%  (\#eq:prop-fc-err)
% \tag*{Fold change in proportion}
\underbrace{\frac{\hat{\text{prop}}_{i}(b)}{\hat{\text{prop}}_{i}(a)}} _\text{estimate}
  &= \frac
    {\text{prop}_{i}(b) \cdot \cancel{\text{efficiency}_{i}} / {\text{mean efficiency}(b)}}
    {\text{prop}_{i}(a) \cdot \cancel{\text{efficiency}_{i}} / {\text{mean efficiency}(a)}}
\\[0.5ex]
  &=
  \underbrace{\frac{\text{prop}_{i}(b)}{\text{prop}_{i}(a)}}_\text{actual}
  \cdot
  \underbrace{\left[\frac{\text{mean efficiency}(b)}{\text{mean efficiency}(a)}\right]^{-1}}_\text{error}
  .
\end{align}

Notes for revision:
- The [[https://da.mikemc.cc/revision/how-taxonomic-bias-affects-differential-abundance-analysis.html#change-between-a-pair-of-samples][current text in 'revision']] can be trimmed into this.
- Footnote: Impact of counting variation, zeros
** Regression analysis of many samples
- P: intro the regression problem
  + State the general regression problem
  + Focus on the specific example of inference of the change in log density with a single covariate $x$ (Equation [[eq:regression]])
- P: Describe impact of bias on regression inference, using results for ordinary least squares (OLS)
  + Variation in the log mean efficiency that is associated with the covariate causes error in the estimated slope; variation that is not associated with the covariate decreases the precision of the estimates.
  + For each species, the statistical bias in covariate coefficient (i.e. slope) = the negative (scaled) covariance between the log mean efficiency and the covariate
    - 3 types of errors: increased magnitude, decreased magnitude, and sign error
    - The relative error is large when the covariance of log mean efficiency with $x$ is large relative to the covariance of the response (log density of species $i$) with $x$.
    - Discuss using figure
  + Even if the slope estimate (or expected estimate) is not affected, the precision can be reduced, leading to missed associations.
- P: (very short). If consistent errors (e.g. ratio-based analysis), then only the intercept is affected, slopes are ok.

Equation stating the basic regression analysis problem
#+name: eq:regression
\begin{align}
  E\left[\log \text{density}_{i} \mid x \right] = a_{0} + a_{1}x
\end{align}

- Note: I'm still thinking through the impact on precision in the case where the residual variation in the log mean efficiency is correlated with that of the log abundance of the focal species (which it generally it will be). In the case where it is not, then the standard error in the estimated slope coefficient will increased in a straightforward way, and whether this is significant depends on the (residual) variance in the log mean efficiency relative to the (residual) variance in log abundance of the focal species.

*** TODO Figure: Illustrate the impact of bias on regression with a simulated example.
* TODO Implications for real-world inference
- See the [[https://mikemc.github.io/differential-abundance-theory/v/0f596f09880fb209cd5a6b87bff16da6d7418ea0/implications-for-real-world-inference.html][Current draft in 'main' branch]]; I'm still working to straighten my thoughts out about how to revise this section
* Potential solutions
Optional figure: Could be helpful to have a conceptual/schematic figure that represents each of these approaches, as a form of visual outline/taxonomy.
** Calibrate compositions using community controls
** Calibrate fold changes using control measurements of targeted taxa
** Use ratio-based estimation methods
** Computational bias-sensitivity analysis
** Bias-aware meta-analysis
