<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>1 Introduction | Taxonomic bias can create spurious results in microbiome differential-abundance analyses unless it is properly accounted for</title>
  <meta name="description" content="1 Introduction | Taxonomic bias can create spurious results in microbiome differential-abundance analyses unless it is properly accounted for" />
  <meta name="generator" content="bookdown 0.22 and GitBook 2.6.7" />

  <meta property="og:title" content="1 Introduction | Taxonomic bias can create spurious results in microbiome differential-abundance analyses unless it is properly accounted for" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="1 Introduction | Taxonomic bias can create spurious results in microbiome differential-abundance analyses unless it is properly accounted for" />
  
  
  

<meta name="author" content="Michael R. McLaren" />
<meta name="author" content="Karen G. Lloyd" />
<meta name="author" content="Benjamin J. Callahan" />


<meta name="date" content="2021-08-04" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="index.html"/>
<link rel="next" href="bias-can-cause-spurious-results-in-proportion-based-da-analysis.html"/>
<script src="libs/header-attrs-2.9/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>



<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="chapter" data-level="1" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="bias-can-cause-spurious-results-in-proportion-based-da-analysis.html"><a href="bias-can-cause-spurious-results-in-proportion-based-da-analysis.html"><i class="fa fa-check"></i><b>2</b> Bias can cause spurious results in proportion-based DA analysis</a>
<ul>
<li class="chapter" data-level="2.1" data-path="bias-can-cause-spurious-results-in-proportion-based-da-analysis.html"><a href="bias-can-cause-spurious-results-in-proportion-based-da-analysis.html#effect-on-regression-of-multiple-samples"><i class="fa fa-check"></i><b>2.1</b> Effect on regression of multiple samples</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="implications-for-real-world-inference.html"><a href="implications-for-real-world-inference.html"><i class="fa fa-check"></i><b>3</b> Implications for real-world inferenceÂ </a>
<ul>
<li class="chapter" data-level="3.1" data-path="implications-for-real-world-inference.html"><a href="implications-for-real-world-inference.html#scenarios-in-which-bias-may-cause-spurious-results"><i class="fa fa-check"></i><b>3.1</b> Scenarios in which bias may cause spurious results</a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="implications-for-real-world-inference.html"><a href="implications-for-real-world-inference.html#error-in-absolute-abundance-measurements"><i class="fa fa-check"></i><b>3.1.1</b> Error in absolute abundance measurements</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="implications-for-real-world-inference.html"><a href="implications-for-real-world-inference.html#we-cannot-simply-ignore-taxonomic-bias"><i class="fa fa-check"></i><b>3.2</b> We cannot simply ignore taxonomic bias</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="potential-solutions.html"><a href="potential-solutions.html"><i class="fa fa-check"></i><b>4</b> Potential solutions</a>
<ul>
<li class="chapter" data-level="4.1" data-path="potential-solutions.html"><a href="potential-solutions.html#using-control-samples-to-calibrate-relative-abundances"><i class="fa fa-check"></i><b>4.1</b> Using control samples to calibrate relative abundances</a></li>
<li class="chapter" data-level="4.2" data-path="potential-solutions.html"><a href="potential-solutions.html#ratio-based-relative-and-absolute-abundance-inference"><i class="fa fa-check"></i><b>4.2</b> Ratio-based relative and absolute abundance inference</a></li>
<li class="chapter" data-level="4.3" data-path="potential-solutions.html"><a href="potential-solutions.html#estimating-error-with-targeted-measurements-of-a-small-number-of-taxa"><i class="fa fa-check"></i><b>4.3</b> Estimating error with targeted measurements of a small number of taxa</a></li>
<li class="chapter" data-level="4.4" data-path="potential-solutions.html"><a href="potential-solutions.html#computational-approaches"><i class="fa fa-check"></i><b>4.4</b> Computational approaches</a></li>
</ul></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="models.html"><a href="models.html"><i class="fa fa-check"></i><b>A</b> Measurement models</a>
<ul>
<li class="chapter" data-level="A.1" data-path="models.html"><a href="models.html#actual-abundances-and-general-notation"><i class="fa fa-check"></i><b>A.1</b> Actual abundances and general notation</a></li>
<li class="chapter" data-level="A.2" data-path="models.html"><a href="models.html#metagenomics-measurement"><i class="fa fa-check"></i><b>A.2</b> Metagenomics measurement</a></li>
<li class="chapter" data-level="A.3" data-path="models.html"><a href="models.html#bulk-absolute-abundance-measurement"><i class="fa fa-check"></i><b>A.3</b> Bulk absolute abundance measurement</a></li>
<li class="chapter" data-level="A.4" data-path="models.html"><a href="models.html#targeted-absolute-abundance-measurement"><i class="fa fa-check"></i><b>A.4</b> Targeted absolute abundance measurement</a></li>
<li class="chapter" data-level="A.5" data-path="models.html"><a href="models.html#spike-ins"><i class="fa fa-check"></i><b>A.5</b> Spike-ins</a></li>
<li class="chapter" data-level="A.6" data-path="models.html"><a href="models.html#complications"><i class="fa fa-check"></i><b>A.6</b> Complications</a>
<ul>
<li class="chapter" data-level="A.6.1" data-path="models.html"><a href="models.html#abundances-units-and-protocol-stages"><i class="fa fa-check"></i><b>A.6.1</b> Abundances units and protocol stages</a></li>
<li class="chapter" data-level="A.6.2" data-path="models.html"><a href="models.html#saturation-in-dna-extraction-yields"><i class="fa fa-check"></i><b>A.6.2</b> Saturation in DNA extraction yields</a></li>
<li class="chapter" data-level="A.6.3" data-path="models.html"><a href="models.html#taxonomic-resolution-of-measurements"><i class="fa fa-check"></i><b>A.6.3</b> Taxonomic resolution of measurements</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="B" data-path="relative.html"><a href="relative.html"><i class="fa fa-check"></i><b>B</b> Differential relative abundance</a>
<ul>
<li class="chapter" data-level="B.1" data-path="relative.html"><a href="relative.html#proportion-based-analyses"><i class="fa fa-check"></i><b>B.1</b> Proportion-based analyses</a>
<ul>
<li class="chapter" data-level="B.1.1" data-path="relative.html"><a href="relative.html#log-proportion"><i class="fa fa-check"></i><b>B.1.1</b> Log proportion</a></li>
<li class="chapter" data-level="B.1.2" data-path="relative.html"><a href="relative.html#log-odds-logit"><i class="fa fa-check"></i><b>B.1.2</b> Log odds (logit)</a></li>
</ul></li>
<li class="chapter" data-level="B.2" data-path="relative.html"><a href="relative.html#ratio-based-analyses"><i class="fa fa-check"></i><b>B.2</b> Ratio-based analyses</a></li>
<li class="chapter" data-level="B.3" data-path="relative.html"><a href="relative.html#higher-order-taxa-formed-by-additive-aggregation"><i class="fa fa-check"></i><b>B.3</b> Higher-order taxa formed by additive aggregation</a></li>
</ul></li>
<li class="chapter" data-level="C" data-path="absolute.html"><a href="absolute.html"><i class="fa fa-check"></i><b>C</b> Differential absolute abundance</a>
<ul>
<li class="chapter" data-level="C.1" data-path="absolute.html"><a href="absolute.html#using-bulk-abundance-measurements"><i class="fa fa-check"></i><b>C.1</b> Using bulk abundance measurements</a>
<ul>
<li class="chapter" data-level="C.1.1" data-path="absolute.html"><a href="absolute.html#shared-bias-components"><i class="fa fa-check"></i><b>C.1.1</b> Shared bias components</a></li>
</ul></li>
<li class="chapter" data-level="C.2" data-path="absolute.html"><a href="absolute.html#using-reference-taxa"><i class="fa fa-check"></i><b>C.2</b> Using reference taxa</a>
<ul>
<li class="chapter" data-level="C.2.1" data-path="absolute.html"><a href="absolute.html#taxa-with-targeted-abundance-measurements"><i class="fa fa-check"></i><b>C.2.1</b> Taxa with targeted abundance measurements</a></li>
<li class="chapter" data-level="C.2.2" data-path="absolute.html"><a href="absolute.html#cellular-spike-ins"><i class="fa fa-check"></i><b>C.2.2</b> Cellular spike-ins</a></li>
<li class="chapter" data-level="C.2.3" data-path="absolute.html"><a href="absolute.html#dna-spike-ins"><i class="fa fa-check"></i><b>C.2.3</b> DNA spike-ins</a></li>
<li class="chapter" data-level="C.2.4" data-path="absolute.html"><a href="absolute.html#taxa-assumed-to-have-a-constant-abundance"><i class="fa fa-check"></i><b>C.2.4</b> Taxa assumed to have a constant abundance</a></li>
<li class="chapter" data-level="C.2.5" data-path="absolute.html"><a href="absolute.html#multiple-reference-taxa"><i class="fa fa-check"></i><b>C.2.5</b> Multiple reference taxa</a></li>
</ul></li>
<li class="chapter" data-level="C.3" data-path="absolute.html"><a href="absolute.html#using-an-equivolumetric-protocol"><i class="fa fa-check"></i><b>C.3</b> Using an equivolumetric protocol</a></li>
<li class="chapter" data-level="C.4" data-path="absolute.html"><a href="absolute.html#computational-methods"><i class="fa fa-check"></i><b>C.4</b> Computational methods</a></li>
</ul></li>
<li class="chapter" data-level="D" data-path="proofs-of-regression-results.html"><a href="proofs-of-regression-results.html"><i class="fa fa-check"></i><b>D</b> Proofs of regression results</a>
<ul>
<li class="chapter" data-level="D.1" data-path="proofs-of-regression-results.html"><a href="proofs-of-regression-results.html#general-regression"><i class="fa fa-check"></i><b>D.1</b> General regression</a></li>
<li class="chapter" data-level="D.2" data-path="proofs-of-regression-results.html"><a href="proofs-of-regression-results.html#linear-least-squares-regression"><i class="fa fa-check"></i><b>D.2</b> Linear least-squares regression</a>
<ul>
<li class="chapter" data-level="D.2.1" data-path="proofs-of-regression-results.html"><a href="proofs-of-regression-results.html#linear-regression-of-microbiome-abundances"><i class="fa fa-check"></i><b>D.2.1</b> Linear regression of microbiome abundances</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="E" data-path="dna-measurement-and-spike-ins.html"><a href="dna-measurement-and-spike-ins.html"><i class="fa fa-check"></i><b>E</b> DNA measurement and spike-ins</a>
<ul>
<li class="chapter" data-level="E.1" data-path="dna-measurement-and-spike-ins.html"><a href="dna-measurement-and-spike-ins.html#expanded-model-and-notation"><i class="fa fa-check"></i><b>E.1</b> Expanded model and notation</a></li>
<li class="chapter" data-level="E.2" data-path="dna-measurement-and-spike-ins.html"><a href="dna-measurement-and-spike-ins.html#targeted-dna-measurement"><i class="fa fa-check"></i><b>E.2</b> Targeted DNA measurement</a></li>
<li class="chapter" data-level="E.3" data-path="dna-measurement-and-spike-ins.html"><a href="dna-measurement-and-spike-ins.html#dna-spike-ins-1"><i class="fa fa-check"></i><b>E.3</b> DNA spike-ins</a></li>
<li class="chapter" data-level="E.4" data-path="dna-measurement-and-spike-ins.html"><a href="dna-measurement-and-spike-ins.html#implications"><i class="fa fa-check"></i><b>E.4</b> Implications</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Taxonomic bias can create spurious results in microbiome differential-abundance analyses unless it is properly accounted for</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="introduction" class="section level1" number="1">
<h1><span class="header-section-number">1</span> Introduction</h1>
<p>Most microbiome research aims to go beyond qualitative descriptions to make quantitative conclusions about associations between specific microbes or community states and key host or environmental properties.
But there is serious disagreement about whether the microbiome measurements made by marker-gene and metagenomic sequencing (MGS) are or can ever be truly quantitative. A primary reason is that taxa vary in how they respond to each step in an MGS protocol, from sample collection to bioinformatic classification.
<!-- TODO: Possible references: Mallick et al., 2017 DOI 10.1186/s13059-017-1359-z; Barlow et al. 2020 https://doi.org/10.1038/s41467-020-16224-6 -->
As a result taxa can differ dramatically (e.g.Â 10-1000X) in how efficiently they are measuredâthat is, converted from cells into taxonomically classified sequencing readsâmaking the relative abundances we observe by MGS inaccurate representations of the actual sample compositions.
Although often associated with variation in primer binding and amplification rates and marker-gene copy-number, large variation in DNA extraction efficiency and in the ability to correctly classify reads among taxa make this taxonomic bias a feature of shotgun metagenomic measurements as much as marker-gene measurements.
Taxonomic bias is protocol specific (<span class="citation"><a href="#ref-mclaren2019cons" role="doc-biblioref">McLaren, Willis, and Callahan</a> (<a href="#ref-mclaren2019cons" role="doc-biblioref">2019</a>)</span>) and can even vary among batches within an experiment (<span class="citation"><a href="#ref-yeh2018taxo" role="doc-biblioref">Yeh et al.</a> (<a href="#ref-yeh2018taxo" role="doc-biblioref">2018</a>)</span>); it thus not only makes MGS measurements inaccurate, but also incomparable between studies.
Taxonomic bias thereby poses a major challenge for any seeking to draw quantitative conclusions from MGS studies.</p>
<p>The fieldâs current answer to this challenge is methods standardization, prioritizing consistency over accuracy.
By treating each sample within a study in exactly the same manner, we can ensure that observed differences between samples are not simply due to methodological differences.
Yet standardization does not guarantee that these observed differences reflect the actual differences in taxonomic composition among samples.
Consider what might be the simplest and most common inference made from MGS: How does the proportion of a taxon vary between samples? An observation that <em>Escherichia</em> coli makes up 10% of the sequencing reads in a fecal sample might, thanks to bias, have arisen from a sample in which the proportion of <em>E. coli</em> is actually 50% or 1%.
Yet many hope that if we observe the proportion of <em>E. coli</em> to increase from 10% to 20% between two samples, its proportion must have actually doubledâby fixing the taxonomic bias of our measurements through standardization, we have (so the argument goes) fixed the measurement error in both samples so that it will cancel when estimate the fold change.
Unfortunately, the results of <span class="citation"><a href="#ref-mclaren2019cons" role="doc-biblioref">McLaren, Willis, and Callahan</a> (<a href="#ref-mclaren2019cons" role="doc-biblioref">2019</a>)</span> show that reality is not so neat.
Because the taxa compete for sequencing effort within a sample, the effect that taxonomic bias has on the observed proportions depends on the taxonomic composition of the sample.
As a result, MGS measurements can lead to spurious inferences about changes across samples even when taxonomic bias is constant for each taxon across samples.</p>
<p>This problem posed by taxonomic bias intersects with a second, growing concern within the microbiome field: that the relative abundances measured by MGS can present a misleading view of absolute-abundance dynamics.
Without making further assumptions (which are often difficult or impossible to verify) one cannot tell whether a doubling in the proportion of a taxon reflects an increase in its abundance or a decrease in the abundances of other taxa.
This concern has motivated the development of various experimental methods to enable researchers to convert the relative abundances measured by MGS into absolute abundances (methods to be reviewed in the Appendix in a future version).
An increasingly popular approach in the study of both environmental and host-associated microbiomes involves making an independent measurement of total microbial abundance in each sample (<span class="citation"><a href="#ref-props2017abso" role="doc-biblioref">Props et al.</a> (<a href="#ref-props2017abso" role="doc-biblioref">2017</a>)</span>,<span class="citation"><a href="#ref-kevorkian2018esti" role="doc-biblioref">Kevorkian et al.</a> (<a href="#ref-kevorkian2018esti" role="doc-biblioref">2018</a>)</span>,<span class="citation"><a href="#ref-lloyd2020evid" role="doc-biblioref">Lloyd et al.</a> (<a href="#ref-lloyd2020evid" role="doc-biblioref">2020</a>)</span>,<span class="citation"><a href="#ref-tettamantiboshier2020comp" role="doc-biblioref">Tettamanti Boshier et al.</a> (<a href="#ref-tettamantiboshier2020comp" role="doc-biblioref">2020</a>)</span>,<span class="citation"><a href="#ref-contijoch2019gutm" role="doc-biblioref">Contijoch et al.</a> (<a href="#ref-contijoch2019gutm" role="doc-biblioref">2019</a>)</span>,<span class="citation"><a href="#ref-vandeputte2017quan" role="doc-biblioref">Vandeputte et al.</a> (<a href="#ref-vandeputte2017quan" role="doc-biblioref">2017</a>)</span>,<span class="citation"><a href="#ref-vieirasilva2019quan" role="doc-biblioref">Vieira-Silva et al.</a> (<a href="#ref-vieirasilva2019quan" role="doc-biblioref">2019</a>)</span>).
This estimate is then multiplied by the MGS proportions to convert them to estimates of absolute abundance.
This simple calibration procedure directly transfers the error in the MGS proportions due to taxonomic bias to the estimated absolute abundances, leading to estimates that are not only individually inaccurate but can imply inaccurate variation in abundance across samples.</p>
<p>Here we summarize recent advances in our understanding of how taxonomic bias affects our ability to accurately determine how taxa vary in relative and absolute abundance across samples from different conditionsâthe inference problem known as <em>differential abundance (DA) analysis</em>.
We first explain how taxonomic bias can lead to spurious DA results and consider the biological conditions in which such spurious results are likely.
But we then describe several experimental and computational methods that can be used to avoid or correct these errors.
Collectively, these methods may provide practical solutions to mitigating the effects of taxonomic bias on DA analysis for a majority of microbiome studies.</p>
</div>
<h3>References</h3>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-contijoch2019gutm" class="csl-entry">
Contijoch, Eduardo J, Graham J Britton, Chao Yang, Ilaria Mogno, Zhihua Li, Ruby Ng, Sean R Llewellyn, et al. 2019. <span>â<span class="nocase">Gut microbiota density influences host physiology and is shaped by host and microbial factors</span>.â</span> <em>Elife</em> 8 (January). <a href="https://doi.org/10.7554/eLife.40553">https://doi.org/10.7554/eLife.40553</a>.
</div>
<div id="ref-kevorkian2018esti" class="csl-entry">
Kevorkian, Richard, Jordan T Bird, Alexander Shumaker, and Karen G Lloyd. 2018. <span>â<span class="nocase">Estimating Population Turnover Rates by Relative Quantification Methods Reveals Microbial Dynamics in Marine Sediment</span>.â</span> <em>Appl. Environ. Microbiol.</em> 84 (1): e01443â17. <a href="https://doi.org/10.1128/AEM.01443-17">https://doi.org/10.1128/AEM.01443-17</a>.
</div>
<div id="ref-lloyd2020evid" class="csl-entry">
Lloyd, Karen G., Jordan T. Bird, Joy Buongiorno, Emily Deas, Richard Kevorkian, Talor Noordhoek, Jacob Rosalsky, and Taylor Roy. 2020. <span>â<span class="nocase">Evidence for a Growth Zone for Deep-Subsurface Microbial Clades in Near-Surface Anoxic Sediments</span>.â</span> <em>Appl. Environ. Microbiol.</em> 86 (19): 1â15. <a href="https://doi.org/10.1128/AEM.00877-20">https://doi.org/10.1128/AEM.00877-20</a>.
</div>
<div id="ref-mclaren2019cons" class="csl-entry">
McLaren, Michael R, Amy D Willis, and Benjamin J Callahan. 2019. <span>â<span class="nocase">Consistent and correctable bias in metagenomic sequencing experiments</span>.â</span> <em>Elife</em> 8 (September): 46923. <a href="https://doi.org/10.7554/eLife.46923">https://doi.org/10.7554/eLife.46923</a>.
</div>
<div id="ref-props2017abso" class="csl-entry">
Props, Ruben, Frederiek-Maarten Kerckhof, Peter Rubbens, Jo De Vrieze, Emma Hernandez Sanabria, Willem Waegeman, Pieter Monsieurs, Frederik Hammes, and Nico Boon. 2017. <span>â<span class="nocase">Absolute quantification of microbial taxon abundances</span>.â</span> <em>ISME J.</em> 11 (2): 584â87. <a href="https://doi.org/10.1038/ismej.2016.117">https://doi.org/10.1038/ismej.2016.117</a>.
</div>
<div id="ref-tettamantiboshier2020comp" class="csl-entry">
Tettamanti Boshier, Florencia A., Sujatha Srinivasan, Anthony Lopez, Noah G. Hoffman, Sean Proll, David N. Fredricks, and Joshua T. Schiffer. 2020. <span>â<span class="nocase">Complementing 16S rRNA Gene Amplicon Sequencing with Total Bacterial Load To Infer Absolute Species Concentrations in the Vaginal Microbiome</span>.â</span> <em>mSystems</em> 5 (2): 1â14. <a href="https://doi.org/10.1128/mSystems.00777-19">https://doi.org/10.1128/mSystems.00777-19</a>.
</div>
<div id="ref-vandeputte2017quan" class="csl-entry">
Vandeputte, Doris, Gunter Kathagen, Kevin Dâhoe, Sara Vieira-Silva, Mireia Valles-Colomer, JoÃ£o Sabino, Jun Wang, et al. 2017. <span>â<span class="nocase">Quantitative microbiome profiling links gut community variation to microbial load</span>.â</span> <em>Nature</em> 551 (7681): 507. <a href="https://doi.org/10.1038/nature24460">https://doi.org/10.1038/nature24460</a>.
</div>
<div id="ref-vieirasilva2019quan" class="csl-entry">
Vieira-Silva, Sara, JoÃ£o Sabino, Mireia Valles-Colomer, Gwen Falony, Gunter Kathagen, Clara Caenepeel, Isabelle Cleynen, Schalk van der Merwe, SÃ©verine Vermeire, and Jeroen Raes. 2019. <span>â<span class="nocase">Quantitative microbiome profiling disentangles inflammation- and bile duct obstruction-associated microbiota alterations across PSC/IBD diagnoses</span>.â</span> <em>Nat. Microbiol.</em> 4 (11): 1826â31. <a href="https://doi.org/10.1038/s41564-019-0483-9">https://doi.org/10.1038/s41564-019-0483-9</a>.
</div>
<div id="ref-yeh2018taxo" class="csl-entry">
Yeh, Yi-Chun, David M. Needham, Ella T. Sieradzki, and Jed A. Fuhrman. 2018. <span>â<span class="nocase">Taxon Disappearance from Microbiome Analysis Reinforces the Value of Mock Communities as a Standard in Every Sequencing Run</span>.â</span> <em>mSystems</em> 3 (3): e00023â18. <a href="https://doi.org/10.1128/mSystems.00023-18">https://doi.org/10.1128/mSystems.00023-18</a>.
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="index.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="bias-can-cause-spurious-results-in-proportion-based-da-analysis.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": false,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/mikemc/differential-abundance-theory/edit/main/article.Rmd",
"text": "Edit"
},
"history": {
"link": "https://github.com/mikemc/differential-abundance-theory/commits/main/article.Rmd",
"text": null
},
"view": {
"link": "https://github.com/mikemc/differential-abundance-theory/blob/main/article.Rmd",
"text": null
},
"download": ["_main.pdf"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
