<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>C Differential absolute abundance | Accurate inference of microbial differential abundance from taxonomically-biased microbiome measurements</title>
  <meta name="description" content="C Differential absolute abundance | Accurate inference of microbial differential abundance from taxonomically-biased microbiome measurements" />
  <meta name="generator" content="bookdown 0.22 and GitBook 2.6.7" />

  <meta property="og:title" content="C Differential absolute abundance | Accurate inference of microbial differential abundance from taxonomically-biased microbiome measurements" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="C Differential absolute abundance | Accurate inference of microbial differential abundance from taxonomically-biased microbiome measurements" />
  
  
  

<meta name="author" content="Michael R. McLaren" />
<meta name="author" content="Karen G. Lloyd" />
<meta name="author" content="Benjamin J. Callahan" />


<meta name="date" content="2021-09-15" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="relative.html"/>
<link rel="next" href="proofs-of-regression-results.html"/>
<script src="libs/header-attrs-2.9/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>
<script type="text/x-mathjax-config">
MathJax.Hub.Config({
TeX: {extensions: ["cancel.js"]}
});
</script>



<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="chapter" data-level="1" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="abundance-measurement.html"><a href="abundance-measurement.html"><i class="fa fa-check"></i><b>2</b> How taxonomic bias affects abundance measurements</a>
<ul>
<li class="chapter" data-level="2.1" data-path="abundance-measurement.html"><a href="abundance-measurement.html#relative-abundances-proportions-and-ratios"><i class="fa fa-check"></i><b>2.1</b> Relative abundances (proportions and ratios)</a></li>
<li class="chapter" data-level="2.2" data-path="abundance-measurement.html"><a href="abundance-measurement.html#absolute-abundances-cell-densities"><i class="fa fa-check"></i><b>2.2</b> Absolute abundances (cell densities)</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="differential-abundance.html"><a href="differential-abundance.html"><i class="fa fa-check"></i><b>3</b> How taxonomic bias affects differential-abundance analysis</a>
<ul>
<li class="chapter" data-level="3.1" data-path="differential-abundance.html"><a href="differential-abundance.html#change-between-a-pair-of-samples"><i class="fa fa-check"></i><b>3.1</b> Change between a pair of samples</a></li>
<li class="chapter" data-level="3.2" data-path="differential-abundance.html"><a href="differential-abundance.html#regression-analysis-of-many-samples"><i class="fa fa-check"></i><b>3.2</b> Regression analysis of many samples</a></li>
<li class="chapter" data-level="3.3" data-path="differential-abundance.html"><a href="differential-abundance.html#regression-error"><i class="fa fa-check"></i><b>3.3</b> Box: Error in estimated regression coefficients and standard errors</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="implications.html"><a href="implications.html"><i class="fa fa-check"></i><b>4</b> Implications for real-world inferenceÂ </a>
<ul>
<li class="chapter" data-level="4.1" data-path="implications.html"><a href="implications.html#systematic-error-in-slope-or-lfc-estimates"><i class="fa fa-check"></i><b>4.1</b> Systematic error in slope or LFC estimates</a></li>
<li class="chapter" data-level="4.2" data-path="implications.html"><a href="implications.html#loss-of-power-and-precision-stub"><i class="fa fa-check"></i><b>4.2</b> Loss of power and precision (stub)</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="solutions.html"><a href="solutions.html"><i class="fa fa-check"></i><b>5</b> Potential solutions</a>
<ul>
<li class="chapter" data-level="5.1" data-path="solutions.html"><a href="solutions.html#calibrate-compositions"><i class="fa fa-check"></i><b>5.1</b> Calibrate compositions using community controls</a></li>
<li class="chapter" data-level="5.2" data-path="solutions.html"><a href="solutions.html#use-ratio-based-abundance-measurement"><i class="fa fa-check"></i><b>5.2</b> Use ratio-based abundance measurement</a></li>
<li class="chapter" data-level="5.3" data-path="solutions.html"><a href="solutions.html#calibrate-fold-changes-using-measurements-of-targeted-taxa"><i class="fa fa-check"></i><b>5.3</b> Calibrate fold changes using measurements of targeted taxa</a></li>
<li class="chapter" data-level="5.4" data-path="solutions.html"><a href="solutions.html#perform-a-bias-sensitivity-analysis"><i class="fa fa-check"></i><b>5.4</b> Perform a bias sensitivity analysis</a></li>
<li class="chapter" data-level="5.5" data-path="solutions.html"><a href="solutions.html#use-bias-aware-meta-analysis"><i class="fa fa-check"></i><b>5.5</b> Use bias-aware meta-analysis</a></li>
</ul></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="models.html"><a href="models.html"><i class="fa fa-check"></i><b>A</b> Measurement models</a>
<ul>
<li class="chapter" data-level="A.1" data-path="models.html"><a href="models.html#actual-abundances-and-general-notation"><i class="fa fa-check"></i><b>A.1</b> Actual abundances and general notation</a></li>
<li class="chapter" data-level="A.2" data-path="models.html"><a href="models.html#metagenomics-measurement"><i class="fa fa-check"></i><b>A.2</b> Metagenomics measurement</a></li>
<li class="chapter" data-level="A.3" data-path="models.html"><a href="models.html#bulk-absolute-abundance-measurement"><i class="fa fa-check"></i><b>A.3</b> Bulk absolute abundance measurement</a></li>
<li class="chapter" data-level="A.4" data-path="models.html"><a href="models.html#targeted-absolute-abundance-measurement"><i class="fa fa-check"></i><b>A.4</b> Targeted absolute abundance measurement</a></li>
<li class="chapter" data-level="A.5" data-path="models.html"><a href="models.html#spike-ins"><i class="fa fa-check"></i><b>A.5</b> Spike-ins</a></li>
<li class="chapter" data-level="A.6" data-path="models.html"><a href="models.html#complications"><i class="fa fa-check"></i><b>A.6</b> Complications</a>
<ul>
<li class="chapter" data-level="A.6.1" data-path="models.html"><a href="models.html#abundances-units-and-protocol-stages"><i class="fa fa-check"></i><b>A.6.1</b> Abundances units and protocol stages</a></li>
<li class="chapter" data-level="A.6.2" data-path="models.html"><a href="models.html#saturation-in-dna-extraction-yields"><i class="fa fa-check"></i><b>A.6.2</b> Saturation in DNA extraction yields</a></li>
<li class="chapter" data-level="A.6.3" data-path="models.html"><a href="models.html#taxonomic-resolution-of-measurements"><i class="fa fa-check"></i><b>A.6.3</b> Taxonomic resolution of measurements</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="B" data-path="relative.html"><a href="relative.html"><i class="fa fa-check"></i><b>B</b> Differential relative abundance</a>
<ul>
<li class="chapter" data-level="B.1" data-path="relative.html"><a href="relative.html#proportion-based-analyses"><i class="fa fa-check"></i><b>B.1</b> Proportion-based analyses</a>
<ul>
<li class="chapter" data-level="B.1.1" data-path="relative.html"><a href="relative.html#log-proportion"><i class="fa fa-check"></i><b>B.1.1</b> Log proportion</a></li>
<li class="chapter" data-level="B.1.2" data-path="relative.html"><a href="relative.html#log-odds-logit"><i class="fa fa-check"></i><b>B.1.2</b> Log odds (logit)</a></li>
</ul></li>
<li class="chapter" data-level="B.2" data-path="relative.html"><a href="relative.html#ratio-based-analyses"><i class="fa fa-check"></i><b>B.2</b> Ratio-based analyses</a></li>
<li class="chapter" data-level="B.3" data-path="relative.html"><a href="relative.html#higher-order-taxa-formed-by-additive-aggregation"><i class="fa fa-check"></i><b>B.3</b> Higher-order taxa formed by additive aggregation</a></li>
</ul></li>
<li class="chapter" data-level="C" data-path="absolute.html"><a href="absolute.html"><i class="fa fa-check"></i><b>C</b> Differential absolute abundance</a>
<ul>
<li class="chapter" data-level="C.1" data-path="absolute.html"><a href="absolute.html#using-bulk-abundance-measurements"><i class="fa fa-check"></i><b>C.1</b> Using bulk abundance measurements</a>
<ul>
<li class="chapter" data-level="C.1.1" data-path="absolute.html"><a href="absolute.html#shared-bias-components"><i class="fa fa-check"></i><b>C.1.1</b> Shared bias components</a></li>
</ul></li>
<li class="chapter" data-level="C.2" data-path="absolute.html"><a href="absolute.html#using-reference-taxa"><i class="fa fa-check"></i><b>C.2</b> Using reference taxa</a>
<ul>
<li class="chapter" data-level="C.2.1" data-path="absolute.html"><a href="absolute.html#taxa-with-targeted-abundance-measurements"><i class="fa fa-check"></i><b>C.2.1</b> Taxa with targeted abundance measurements</a></li>
<li class="chapter" data-level="C.2.2" data-path="absolute.html"><a href="absolute.html#cellular-spike-ins"><i class="fa fa-check"></i><b>C.2.2</b> Cellular spike-ins</a></li>
<li class="chapter" data-level="C.2.3" data-path="absolute.html"><a href="absolute.html#dna-spike-ins"><i class="fa fa-check"></i><b>C.2.3</b> DNA spike-ins</a></li>
<li class="chapter" data-level="C.2.4" data-path="absolute.html"><a href="absolute.html#taxa-assumed-to-have-a-constant-abundance"><i class="fa fa-check"></i><b>C.2.4</b> Taxa assumed to have a constant abundance</a></li>
<li class="chapter" data-level="C.2.5" data-path="absolute.html"><a href="absolute.html#multiple-reference-taxa"><i class="fa fa-check"></i><b>C.2.5</b> Multiple reference taxa</a></li>
</ul></li>
<li class="chapter" data-level="C.3" data-path="absolute.html"><a href="absolute.html#using-an-equivolumetric-protocol"><i class="fa fa-check"></i><b>C.3</b> Using an equivolumetric protocol</a></li>
<li class="chapter" data-level="C.4" data-path="absolute.html"><a href="absolute.html#computational-methods"><i class="fa fa-check"></i><b>C.4</b> Computational methods</a></li>
</ul></li>
<li class="chapter" data-level="D" data-path="proofs-of-regression-results.html"><a href="proofs-of-regression-results.html"><i class="fa fa-check"></i><b>D</b> Proofs of regression results</a>
<ul>
<li class="chapter" data-level="D.1" data-path="proofs-of-regression-results.html"><a href="proofs-of-regression-results.html#general-regression"><i class="fa fa-check"></i><b>D.1</b> General regression</a></li>
<li class="chapter" data-level="D.2" data-path="proofs-of-regression-results.html"><a href="proofs-of-regression-results.html#linear-least-squares-regression"><i class="fa fa-check"></i><b>D.2</b> Linear least-squares regression</a>
<ul>
<li class="chapter" data-level="D.2.1" data-path="proofs-of-regression-results.html"><a href="proofs-of-regression-results.html#linear-regression-of-microbiome-abundances"><i class="fa fa-check"></i><b>D.2.1</b> Linear regression of microbiome abundances</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="E" data-path="dna-measurement-and-spike-ins.html"><a href="dna-measurement-and-spike-ins.html"><i class="fa fa-check"></i><b>E</b> DNA measurement and spike-ins</a>
<ul>
<li class="chapter" data-level="E.1" data-path="dna-measurement-and-spike-ins.html"><a href="dna-measurement-and-spike-ins.html#expanded-model-and-notation"><i class="fa fa-check"></i><b>E.1</b> Expanded model and notation</a></li>
<li class="chapter" data-level="E.2" data-path="dna-measurement-and-spike-ins.html"><a href="dna-measurement-and-spike-ins.html#targeted-dna-measurement"><i class="fa fa-check"></i><b>E.2</b> Targeted DNA measurement</a></li>
<li class="chapter" data-level="E.3" data-path="dna-measurement-and-spike-ins.html"><a href="dna-measurement-and-spike-ins.html#dna-spike-ins-1"><i class="fa fa-check"></i><b>E.3</b> DNA spike-ins</a></li>
<li class="chapter" data-level="E.4" data-path="dna-measurement-and-spike-ins.html"><a href="dna-measurement-and-spike-ins.html#implications-1"><i class="fa fa-check"></i><b>E.4</b> Implications</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Accurate inference of microbial differential abundance from taxonomically-biased microbiome measurements</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="absolute" class="section level1" number="8">
<h1><span class="header-section-number">C</span> Differential absolute abundance</h1>
<p>This section considers various ways to estimate absolute abundances of individual taxa for the purposes of estimating log fold changes (LFCs) in absolute abundance, or more generically of performing linear regression on log absolute abundance.</p>
<div id="using-bulk-abundance-measurements" class="section level2" number="8.1">
<h2><span class="header-section-number">C.1</span> Using bulk abundance measurements</h2>
<p>The absolute abundance of taxon <span class="math inline">\(i\)</span> is estimated by multiplying its metagenomics proportion by the bulk abundance measurement,
<span class="math display" id="eq:A-hat-bulk">\[\begin{align}
  \tag{C.1}
  \hat A_{ij} &amp;= m_{ij} K_j.
\end{align}\]</span>
The error in this estimate due to the experimental bias in <span class="math inline">\(M\)</span> and <span class="math inline">\(K\)</span> is given by substituting equations <a href="models.html#eq:m">(A.3)</a> and <a href="models.html#eq:K">(A.4)</a> into <a href="absolute.html#eq:A-hat-bulk">(C.1)</a>, giving
<span class="math display" id="eq:error-bulk">\[\begin{align}
  \tag{C.2}
  \hat A_{ij} &amp;= \frac{a_{ij} B^{(M)}_i}{\bar B^{(M)}_j} \cdot A_{Tj} \bar B^{(K)}_j
  \\&amp;= A_{ij} \cdot \frac{B^{(M)}_i \bar B^{(K)}_j}{\bar B^{(M)}_j}.
\end{align}\]</span>
The fold error in the estimated abundance of a particular taxon in a particular sample thus equals the metagenomics efficiency of that taxon times the ratio of the sample mean efficiencies of the bulk measurement to the metagenomics measurement.
Next I consider the error in the estimated fold change in <span class="math inline">\(A_i\)</span> between two samples <span class="math inline">\(j\)</span> and <span class="math inline">\(j&#39;\)</span>.
As in the case of log proportion estimates, the taxon-specific, sample-independent error term (<span class="math inline">\(B^{(M)}_i\)</span>) cancels, but the taxon-independent, sample-specific error term (<span class="math inline">\({\bar B^{(K)}_j}/{\bar B^{(M)}_j}\)</span>) does not, giving
<span class="math display" id="eq:fc-bulk">\[\begin{align}
  \tag{C.3}
  \frac{\hat A_{ij&#39;}}{\hat A_{ij}}
  &amp;= \frac{A_{ij&#39;}}{A_{ij}} 
  \cdot \frac{\bar B^{(M)}_j \bar B^{(K)}_{j&#39;}}{\bar B^{(M)}_{j&#39;} \bar B^{(K)}_j}.
  \\&amp;= \frac{A_{ij&#39;}}{A_{ij}} 
  \cdot \frac{\bar B^{(K)}_{j&#39;} / \bar B^{(M)}_{j&#39;}}{\bar B^{(K)}_j / \bar B^{(M)}_j }.
\end{align}\]</span>
In words, the fold error in the estimated fold change in the absolute abundance of any taxon equals the fold change in the ratio of the mean efficiency of the bulk abundance measurement to that of the metagenomics measurement.</p>
<p>TODO: Add regression result.</p>
<div id="shared-bias-components" class="section level3" number="8.1.1">
<h3><span class="header-section-number">C.1.1</span> Shared bias components</h3>
<p>Motivation: Consider an experiment in which the community has been profiled by 16S sequencing and bulk abundance has been estimated by 16S qPCR.
The taxon-specific bias associated with each of these measurements may be highly similar; both are measuring 16S copies in the same aliquot of already-extracted DNA, which has already been affected by DNA extraction bias and 16S copy-number variation.
If the same primers are used then PCR bias might also be shared.
How does having a large amount of bias being shared between the bulk and sequencing measurement affect the error in abundance estimates in individual samples and the fold changes between samples?</p>
<p>In this case, the ratio of mean efficiencies, <span class="math inline">\(\bar B^{(K)}_j / \bar B^{(M)}_j\)</span>, will vary much less across samples than either mean efficiency itself, such that the error term in <a href="absolute.html#eq:fc-bulk">(C.3)</a> will be small.
Therefore the estimated fold changes may be accurate despite potentially large variation in the metagenomics sample mean efficiency across samples.
Note that the estimated absolute abundance in the individual sample, <a href="absolute.html#eq:error-bulk">(C.2)</a>, remains inaccurate since the unknown taxon-specific efficiency remains.</p>
<p>TODO: Add derivation, where we formally define <span class="math inline">\(B^{(M)}\)</span> and <span class="math inline">\(B^{(K)}\)</span> in terms of shared and non-shared components, and show that the variation in the ratio of mean efficiencies reduces to that of the non-shared components.</p>
</div>
</div>
<div id="using-reference-taxa" class="section level2" number="8.2">
<h2><span class="header-section-number">C.2</span> Using reference taxa</h2>
<p>TODO: Consider rewriting to be in terms of the ratio of counts * the reference abundance, to emphasize that this is a ratio-based method.</p>
<div id="taxa-with-targeted-abundance-measurements" class="section level3" number="8.2.1">
<h3><span class="header-section-number">C.2.1</span> Taxa with targeted abundance measurements</h3>
<p>First consider the case of just a single reference taxon, <span class="math inline">\(r\)</span>, for which we have targeted measurements.
We can estimate the absolute abundances of all taxa by scaling the read counts by the ratio of the abundance measurement to read count for the reference taxon,
<span class="math display" id="eq:A-hat-targeted">\[\begin{align}
  \tag{C.4}
  \hat A_{ij} &amp;= M_{ij} \cdot \frac{T_{rj}}{M_{rj}}.
\end{align}\]</span>
Accounting for systematic error in both the metagenomcs measurement <a href="models.html#eq:M">(A.1)</a> and the targeted measurement <a href="models.html#eq:T">(A.5)</a> gives
<span class="math display" id="eq:error-targeted" id="eq:error-targeted">\[\begin{align}
  \tag{C.5}
  \hat A_{ij} &amp;= A_{ij} \cdot \frac{B^{(M)}_i B^{(T)}_r}{B^{(M)}_r}.
\end{align}\]</span>
Since the error in \tag{C.5} is sample-independent, it cancels out when we use this equation to compute the fold change between two samples,
<span class="math display" id="eq:fc-targeted">\[\begin{align}
  \tag{C.6}
  \frac{\hat A_{ij&#39;}}{\hat A_{ij}} &amp;=\frac{\hat A_{ij&#39;}}{\hat A_{ij}}.
\end{align}\]</span>
Thus our model predicts that normalizing read counts to a reference taxon, while providing systematically-distorted abundance estimates, provides accurate fold-change estimates.</p>
</div>
<div id="cellular-spike-ins" class="section level3" number="8.2.2">
<h3><span class="header-section-number">C.2.2</span> Cellular spike-ins</h3>
<p>Spike-ins are a way to create a reference taxon whose abundance is known by design and so doesnât need to be measured.
Typically the spike-in is added in a fixed concentration to all samples; however, this is unnecessary, so long as we know the amount it was added.
In either case we can use <a href="absolute.html#eq:A-hat-targeted">(C.4)</a> to estimate the abundances of all taxa from the spike-in taxa, substituting the spike-in abundance <span class="math inline">\(S_{ir}\)</span> for <span class="math inline">\(T_{ir}\)</span>,
Our model predicts that normalizing read counts to a spike-in taxon gives systematically-distorted abundance estimates but accurate fold-change estimates for the same reason as in the targeted case.</p>
</div>
<div id="dna-spike-ins" class="section level3" number="8.2.3">
<h3><span class="header-section-number">C.2.3</span> DNA spike-ins</h3>
</div>
<div id="taxa-assumed-to-have-a-constant-abundance" class="section level3" number="8.2.4">
<h3><span class="header-section-number">C.2.4</span> Taxa assumed to have a constant abundance</h3>
<p>By this approach, we identify one or more taxa that we assume to have a constant absolute abundance across samples.
Inference proceeds as in the targeted and spike-in case, using <a href="absolute.html#eq:A-hat-targeted">(C.4)</a> but with the reference abundance <span class="math inline">\(T_r\)</span> set to an arbitrary constant as we may not know the true abundance.
In this case, the effect of bias cancels in fold-change calculations for the same reasons as the targeted and spike-in cases.
Note: The reference taxon could be the host.</p>
</div>
<div id="multiple-reference-taxa" class="section level3" number="8.2.5">
<h3><span class="header-section-number">C.2.5</span> Multiple reference taxa</h3>
<p>Under the deterministic model we consider here, having multiple reference taxa in the same sample is completely redundant - we learn no new information.
In reality, having information from multiple taxa from a single sample should allow us to make more precise estimates, provided that we use them in a way so that the effects of bias do not offset any reductions in noise.
How might we leverage measurements of multiple reference taxa, <span class="math inline">\(r \in R\)</span>, in a way that remains robust to bias?</p>
<p>First consider the situation where we can assume that all reference taxa are in each sample, as would typically be the case for a spike-in.
One possibility is to sum the metagenomic and targeted abundances from the different taxa,
<span class="math display" id="eq:A-hat-targeted-sum">\[\begin{align}
  \tag{C.7}
  \hat A_{ij} &amp;= M_{ij} \cdot \frac{\sum_{r \in R} T_{rj}}{\sum_{r \in R} M_{rj}}.
\end{align}\]</span>
To see this, let <span class="math inline">\(A_{Rj} = \sum_{r \in R} A_{rj}\)</span> be the summed actual abundances of the reference taxa.
Similarly define <span class="math inline">\(M_{Rj} = \sum_{r \in R} M_{rj}\)</span> and <span class="math inline">\(T_{Rj} = \sum_{r \in R} T_{rj}\)</span> be the summed read counts and targeted measurements.
Now
<span class="math display">\[\begin{align}
  \hat A_{ij} &amp;= M_{ij} \cdot \frac{T_{Rj}}{M_{Rj}}.
\end{align}\]</span>
Also let <span class="math inline">\(\bar B^{(M)}_{Rj} = \sum_{r \in R} B^{(M)}_{rj} A_{rj} / A_{Rj}\)</span> be the mean metagenomics efficiency among the reference taxa and similarly define <span class="math inline">\(\bar B^{(T)}_{Rj}\)</span> as the mean targeted efficiency.
A little algebra shows that
<span class="math display" id="eq:T-Rj">\[\begin{align}
  \tag{A.6}
  T_{Rj} 
    &amp;= A_{Rj} \cdot \sum_{r \in R} \frac{A_{rj}}{A_{Rj}} \bar B^{(T)}_{Rj}
  \\&amp;= A_{Rj} \bar B^{(T)}_{Rj}
\end{align}\]</span>
while
<span class="math display" id="eq:M-Rj">\[\begin{align}
  \tag{C.8}
  M_{Rj} &amp;= A_{Rj} \bar B^{(M)}_{Rj} C_j.
\end{align}\]</span>
And since <span class="math inline">\(M_{ij} = A_{ij} B^{(M)}_{i} C_j\)</span> <a href="models.html#eq:M">(A.1)</a> we can write the estimated <a href="absolute.html#eq:A-hat-targeted-sum">(C.7)</a> abundance of taxon <span class="math inline">\(i\)</span> as
<span class="math display" id="eq:error-targeted-sum">\[\begin{align}
  \tag{C.9}
  \hat A_{ij} &amp;= A_{ij} \cdot \frac{B^{(M)}_i \bar B^{(T)}_{Rj}}{\bar B^{(M)}_{Rj}}.
\end{align}\]</span>
The error now depends on relative abundances among the reference taxa through their mean bias values.
Therefore, fold changes estimated using this approach are not necessarily robust to bias if the relative abundances among the reference taxa vary.
Spike-in experiments are typically designed so that the reference taxa typically have the same relative abundances with respect to each other in all samples, in which case this method would remain robust to bias.</p>
<p>Another approach is to multiply by the geometric mean abundances of the taxa,
<span class="math display" id="eq:A-hat-targeted-prod">\[\begin{align}
  \tag{C.10}
  \hat A_{ij} &amp;= M_{ij} \cdot \left[\frac{\prod_{r \in R} T_{rj}}{\prod_{r \in R} M_{rj}}\right]^{1/|R|}.
\end{align}\]</span>
The error in this case is
<span class="math display" id="eq:error-targeted-prod">\[\begin{align}
  \tag{C.11}
  \hat A_{ij} 
  &amp;= A_{ij} B^{(M)}_r C_j \cdot 
    \left[\frac{\prod_{r \in R} A_{rj} B^{(T)}_r} {\prod_{r \in R} A_{rj} B^{(M)}_r C_j} \right]^{1/|R|} \\
  &amp;= A_{ij} \cdot 
    \left[\frac{\prod_{r \in R} B^{(T)}_r} {\prod_{r \in R} B^{(M)}_r} \right]^{1/|R|}.
\end{align}\]</span>
The multiplicative error is now constant and will cancel in fold-change calculations even if the reference taxa vary in their relative abundances.</p>
<p>Thus the geometric-mean method appears to be robust to bias, but as written it requires that all reference taxa are present in all samples, whereas the summation approach is sensitive to bias but only requires at least one reference taxon to be present in the sample.</p>
<p>Now consider the case where different reference taxa are present (above our detection limit) in each sample, as might be the case for naturally occuring reference taxa.
We can modify the geometric-mean approach to work (so long as at least one reference is present in each sample) as follows.
From the samples where multiple references are present, we are able to measure the differnetial bias <span class="math inline">\(B^{(M/T)}_r / B^{(M/T)}_{r&#39;}\)</span> among the references <span class="math inline">\(r, r&#39;\)</span>.
From measurement of reference <span class="math inline">\(r&#39;\)</span>, we know the value of <span class="math inline">\(T_{r&#39;j} / M_{r&#39;j}\)</span>, which combined with the differential bias tells us what the ratio <span class="math inline">\(T_{rj} / M_{rj}\)</span> should equal, namely
<span class="math display">\[\begin{align}
  \frac{T_{rj}}{M_{rj}} = \frac{T_{r&#39;j}}{M_{r&#39;j}} \cdot \frac{B^{(M/T)}_r}{B^{(M/T)}_{r&#39;}}.
\end{align}\]</span>
Therefore, in samples where <span class="math inline">\(r\)</span> is missing but <span class="math inline">\(r&#39;\)</span> is present, we can replace
<span class="math inline">\(\frac{T_{rj}}{M_{rj}}\)</span> in the product with its predicted value.
(More generally, we can use the geometric mean of the values predicted by each of the reference taxa that are present).</p>
<p>This imputation procedure is a simple plug-in approach to show that it is possible to use a multiplicative approach to using multiple references even when the references vary in their presence among samples.
Real inference should ideally be done with a fully generative statistical model that includes bias as well as noise-generating processes and can be used for maximum likelihood or Bayesian inference. In such models, the imputation might be implicitly handled as part of the likelihood, and more generally the values <span class="math inline">\(\frac{T_{rj}}{M_{rj}}\)</span> for various <span class="math inline">\(r\)</span> can be naturally and automatically weighted (in terms of their influence on <span class="math inline">\(\hat A_{ij}\)</span>) according to their predicted precision by the model fitting procedure.</p>
<!-- TODO: Move the definitions and relevant intermediate results to models.Rmd and  trim the above derivations, including for the aggregates and the differential bias among multiple reference taxa. -->
</div>
</div>
<div id="using-an-equivolumetric-protocol" class="section level2" number="8.3">
<h2><span class="header-section-number">C.3</span> Using an equivolumetric protocol</h2>
</div>
<div id="computational-methods" class="section level2" number="8.4">
<h2><span class="header-section-number">C.4</span> Computational methods</h2>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="relative.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="proofs-of-regression-results.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": false,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/mikemc/differential-abundance-theory/edit/main/absolute-abundance.Rmd",
"text": "Edit"
},
"history": {
"link": "https://github.com/mikemc/differential-abundance-theory/commits/main/absolute-abundance.Rmd",
"text": null
},
"view": {
"link": "https://github.com/mikemc/differential-abundance-theory/blob/main/absolute-abundance.Rmd",
"text": null
},
"download": ["_main.pdf"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
